File: gestion_personnel_front/src/app/models/employee/employee-form/employee-form.ts
export class EmployeeForm {
    id: number;
    secondname: string;
    firstname: string;
    placeOfBirth: string;
    dateOfBirth: Date;

    email: string;
    address: string;
    postalCode: number;
    city: string;
    homenumber: number;
    phonenumber: number;

    post: string;
    dateOfHiring: Date;
    dateEndOfContract: Date;
    salary: number;
    hoursPerWeekId: number;
    contractId: number;
    departmentId: number;

    constructor(
        id: number,
        secondname: string,
        firstname: string,
        placeOfBirth: string,
        dateOfBirth: Date,

        email: string,
        address: string,
        postalCode: number,
        city: string,
        homenumber: number,
        phonenumber: number,

        post: string,
        dateOfHiring: Date,
        dateEndOfContract: Date,
        salary: number,
        hoursPerWeekId: number,
        contractId: number,
        departmentId: number
    ) {
        this.id = id;
        this.secondname = secondname;
        this.firstname = firstname;
        this.placeOfBirth = placeOfBirth;
        this.dateOfBirth = dateOfBirth;

        this.email = email;
        this.address = address;
        this.postalCode = postalCode;
        this.city = city;
        this.homenumber = homenumber;
        this.phonenumber = phonenumber;

        this.post = post;
        this.dateOfHiring = dateOfHiring;
        this.dateEndOfContract = dateEndOfContract;
        this.salary = salary;
        this.hoursPerWeekId = hoursPerWeekId;
        this.contractId =  contractId;
        this.departmentId = departmentId;
    }
}

File: gestion_personnel/src/main/java/com/cdaprojet/gestion_personnel/model/employeeModel/employee/EmployeeForm.java
package com.cdaprojet.gestion_personnel.model.employeeModel.employee;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class EmployeeForm {
    
    private long id;
    private String secondname;
    private String firstname;
    private String placeOfBirth;
    private LocalDate dateOfBirth;

    private String email;
    private String address;
    private int postalCode;
    private String city;
    private int homenumber;
    private int phonenumber;

    private String post;
    private LocalDate dateOfHiring;
    private LocalDate dateEndOfContract;
    private long salary;
    private long hoursPerWeekId;
    private long contractId;
    private long departmentId;
}
